src/Processors/Formats/Impl/IonRowOutputFormat.cpp

```diff
@@ -0,0 +1,323 @@
+            writer->writeString(assert_cast<const ColumnString &>(column).getDataAt(row_num).toView());
+            return;
+        }
+        case TypeIndex::FixedString: {
+            writer->writeString(assert_cast<const ColumnFixedString &>(column).getDataAt(row_num).toView());
+            return;
+        }
+        case TypeIndex::IPv6: {
+            writer->writeIPv6(assert_cast<const ColumnIPv6 &>(column).getElement(row_num));
+            return;
+        }
+        case TypeIndex::Array: {
+            auto nested_type = assert_cast<const DataTypeArray &>(*data_type).getNestedType();
+            const ColumnArray & column_array = assert_cast<const ColumnArray &>(column);
+            const IColumn & nested_column = column_array.getData();
+            const ColumnArray::Offsets & offsets = column_array.getOffsets();
+            size_t offset = offsets[row_num - 1];
+            size_t size = offsets[row_num] - offset;
+            writer->writeStartList();
+            for (size_t i = 0; i < size; ++i)
+                serializeField(nested_column, nested_type, offset + i);
+            writer->writeFinishList();
+            return;
+        }
+        case TypeIndex::Tuple: {
+            const auto & tuple_type = assert_cast<const DataTypeTuple &>(*data_type);
+            const auto & nested_types = tuple_type.getElements();
+            const ColumnTuple & column_tuple = assert_cast<const ColumnTuple &>(column);
+            const auto & nested_columns = column_tuple.getColumns();
+            writer->writeStartList();
+            for (size_t i = 0; i < nested_types.size(); ++i)
+                serializeField(*nested_columns[i], nested_types[i], row_num);
+            writer->writeFinishList();
+            return;
+        }
+        case TypeIndex::Nullable: {
+            auto nested_type = removeNullable(data_type);
+            const ColumnNullable & column_nullable = assert_cast<const ColumnNullable &>(column);
+            if (!column_nullable.isNullAt(row_num))
+                serializeField(column_nullable.getNestedColumn(), nested_type, row_num);
+            else if (isInteger(nested_type))
+                writer->writeTypedNull(NullableIonDataType::Integer);
+            else if (isFloat(nested_type))
+                writer->writeTypedNull(NullableIonDataType::Float);
+            else if (isDecimal(nested_type))
+                writer->writeTypedNull(NullableIonDataType::Decimal);
+            else if (isDateOrDate32OrDateTimeOrDateTime64(nested_type))
+                writer->writeTypedNull(NullableIonDataType::DateTime);
+            else if (isStringOrFixedString(nested_type) || isIPv4(nested_type) || isIPv6(nested_type))
+                writer->writeTypedNull(NullableIonDataType::String);
+            else
+                writer->writeNull();
+            return;
+        }
+        case TypeIndex::Nothing: {
+            writer->writeNull();
+            return;
+        }
```

Comment:
This is unnecessary, it's impossible to have `Nothing` data type in `OutputFormat`